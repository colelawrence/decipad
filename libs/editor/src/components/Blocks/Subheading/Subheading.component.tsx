import styled from '@emotion/styled';
import { SlatePluginComponent } from '@udecode/slate-plugins-core';

const Element = styled('h3')({
  fontSize: '20px',
  margin: 0,
  padding: '1rem 0 0.5rem 0',
  fontWeight: 'bold',
  color: '#121214',
  '& ::selection': {
    backgroundColor: 'rgba(196, 202, 251, 0.5)',
  },
});

export const Subheading: SlatePluginComponent = ({ attributes, children }) => {
  return <Element {...attributes}>{children}</Element>;
};
