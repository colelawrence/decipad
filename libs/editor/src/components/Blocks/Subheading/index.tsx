import styled from '@emotion/styled';
import { SlatePluginComponent } from '@udecode/slate-plugins-core';
import React from 'react';

const Element = styled('h3')({
  fontSize: '20px',
  margin: 0,
  padding: '1rem 0 0.5rem 0',
  fontWeight: 'bold',
  color: '#121214',
});

export const Subheading: SlatePluginComponent = ({ attributes, children }) => {
  return <Element {...attributes}>{children}</Element>;
};
