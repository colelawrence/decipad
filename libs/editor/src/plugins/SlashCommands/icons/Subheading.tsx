import { FC } from 'react';

export const Subheading = (): ReturnType<FC> => {
  return (
    <svg
      width="19"
      height="12"
      viewBox="0 0 19 12"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M2.84375 10.9989L3.74396 8.30895H7.84553L8.75107 10.9989H10.8604L7.01456 0.0898438H4.57493L0.734375 10.9989H2.84375ZM4.27663 6.72159L5.75213 2.32706H5.83736L7.31286 6.72159H4.27663Z"
        fill="#121214"
      />
      <path
        d="M14.3665 11.1641C15.6502 11.1641 16.4173 10.5621 16.7688 9.875H16.8328V10.9989H18.6865V5.52308C18.6865 3.36044 16.9233 2.71058 15.3626 2.71058C13.6421 2.71058 12.321 3.47763 11.8949 4.9691L13.6953 5.22479C13.8871 4.66548 14.4304 4.18608 15.3732 4.18608C16.2681 4.18608 16.7582 4.64418 16.7582 5.44851V5.48047C16.7582 6.03445 16.1776 6.06108 14.734 6.21555C13.1467 6.38601 11.6286 6.86009 11.6286 8.70312C11.6286 10.3118 12.8058 11.1641 14.3665 11.1641ZM14.8672 9.74716C14.0629 9.74716 13.4876 9.37962 13.4876 8.67116C13.4876 7.93075 14.1321 7.6218 14.995 7.49929C15.5011 7.43004 16.5132 7.3022 16.7635 7.09979V8.06392C16.7635 8.97479 16.0284 9.74716 14.8672 9.74716Z"
        fill="#121214"
      />
    </svg>
  );
};
