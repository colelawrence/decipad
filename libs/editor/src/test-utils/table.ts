import {
  ELEMENT_TABLE,
  ELEMENT_TH,
  ELEMENT_TR,
  ELEMENT_TD,
} from '@udecode/plate';
import {
  InteractiveTable,
  TableCellType,
  Td,
  Th,
} from '../plugins/InteractiveTable/table';
import {
  ELEMENT_TABLE_CAPTION,
  ELEMENT_THEAD,
  ELEMENT_HEAD_TR,
  ELEMENT_TBODY,
} from '../utils/elementTypes';

/**
 * Creates a row with 1 named column and {@param numberOfRows} rows of non-empty cells.
 */
export const createTable = (numberOfRows: number): InteractiveTable => ({
  type: ELEMENT_TABLE,
  children: [
    {
      type: ELEMENT_TABLE_CAPTION,
      children: [{ text: 'Table' }],
    },
    {
      type: ELEMENT_THEAD,
      children: [
        {
          type: ELEMENT_HEAD_TR,
          children: [
            {
              type: ELEMENT_TH,
              attributes: { cellType: 'string' },
              children: [{ text: 'Col 1' }],
            },
          ],
        },
      ],
    },
    {
      type: ELEMENT_TBODY,
      children: Array.from({ length: numberOfRows }, (_, i) => ({
        type: ELEMENT_TR,
        children: [
          {
            type: ELEMENT_TD,
            attributes: { cellType: 'string' },
            children: [{ text: `Col 1 Row ${i + 1}` }],
          },
        ],
      })),
    },
  ],
});

export const createTableWithTypeAndData = (
  cellType: TableCellType,
  data: string[]
): InteractiveTable => ({
  type: ELEMENT_TABLE,
  children: [
    {
      type: ELEMENT_TABLE_CAPTION,
      children: [{ text: 'Table' }],
    },
    {
      type: ELEMENT_THEAD,
      children: [
        {
          type: ELEMENT_HEAD_TR,
          children: [
            {
              type: ELEMENT_TH,
              attributes: { cellType },
              children: [{ text: 'Col 1' }],
            },
          ],
        },
      ],
    },
    {
      type: ELEMENT_TBODY,
      children: data.map((text) => ({
        type: ELEMENT_TR,
        children: [
          {
            type: ELEMENT_TD,
            attributes: { cellType },
            children: [{ text }],
          },
        ],
      })),
    },
  ],
});

export const printTable = ({
  children: [_, thead, tbody],
}: InteractiveTable): string[] => {
  const tdsToStrings = (tds: (Th | Td)[]) =>
    tds.map((td) => td.children[0].text).join(' | ');

  return [
    tdsToStrings(thead.children[0].children),
    ...tbody.children.map((tr) => tdsToStrings(tr.children)),
  ];
};
