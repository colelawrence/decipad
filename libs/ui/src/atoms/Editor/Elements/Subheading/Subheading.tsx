import { css } from '@emotion/react';
import { PlatePluginComponent } from '@udecode/plate';
import { h2 } from '../../../../primitives';

const styles = css(h2, {
  padding: '16px 0',
});

export const SubheadingElement: PlatePluginComponent = ({
  attributes,
  children,
}) => {
  return (
    <h3 css={styles} {...attributes}>
      {children}
    </h3>
  );
};
