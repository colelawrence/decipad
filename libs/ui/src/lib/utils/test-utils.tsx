import { ChakraProvider } from '@chakra-ui/react';
import { render, RenderOptions } from '@testing-library/react';
import React from 'react';

import { Type } from '@decipad/language';
import { theme, ResultsContextProvider } from '../..';

const results = {
  test: {
    type: Type.build({
      type: 'number',
      unit: {
        unit: 'apples',
      },
    } as any),
    value: [20],
    errors: [],
  },
  noResult: null,
};

const AllTheProviders: React.FC = ({ children }) => {
  return (
    <ChakraProvider resetCSS theme={theme}>
      <ResultsContextProvider value={results}>
        {children}
      </ResultsContextProvider>
    </ChakraProvider>
  );
};

const customRender = (
  ui: JSX.Element,
  options?: Omit<RenderOptions, 'queries'>
) => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';
// override render method
export { customRender as render };
