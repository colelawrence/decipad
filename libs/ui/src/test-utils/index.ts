export * from './menu';
export * from './language';
