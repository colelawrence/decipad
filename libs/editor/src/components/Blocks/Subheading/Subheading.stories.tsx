import React from 'react';
import { Subheading } from './Subheading.component';

export default {
  title: 'Editor/Blocks/Subheading',
  component: Subheading,
};

export const Default = (args: { text: string }) => (
  <Subheading
    leaf={{ attributes: {}, text: '' }}
    element={{ children: [{ text: '' }] }}
    text={{ text: '' }}
    attributes={{
      'data-slate-node': 'element',
      ref: null,
      'data-slate-leaf': true,
    }}
  >
    {args.text}
  </Subheading>
);

Default.args = {
  text: 'Subheading',
};

Default.storyName = 'Subheading';
