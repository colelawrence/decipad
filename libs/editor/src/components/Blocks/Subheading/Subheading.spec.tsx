import React from 'react';
import { render } from 'test-utils';
import { Default } from './Subheading.stories';

describe('Subheading Block', () => {
  it('matches snapshot', () => {
    const { container } = render(<Default text="Subheading" />);
    expect(container).toMatchSnapshot();
  });
});
