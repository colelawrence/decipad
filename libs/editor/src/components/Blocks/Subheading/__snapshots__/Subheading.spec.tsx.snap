// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subheading Block matches snapshot 1`] = `
.emotion-0 {
  font-size: 20px;
  margin: 0;
  padding: 1rem 0 0.5rem 0;
  font-weight: bold;
  color: #121214;
}

<div>
  <h3
    class="emotion-0"
    data-slate-leaf="true"
    data-slate-node="element"
  >
    Subheading
  </h3>
</div>
`;
